src/JMS/Serializer/YamlSerializationVisitor.php

```diff
@@ -111,6 +111,11 @@ public function visitArray($data, array $type, Context $context)
                 ->rtrim(false)
                 ->writeln(' {}')
             ;
+        } elseif (0 === count($data)) {
```

Comment:
I'm not sure `empty($data)` is the good way. We check for an empty array.

All another empty values should have `~` and not `[]`.